data_steward/cdr_cleaner/cleaning_rules/base_cleaning_rule.py

```diff
@@ -276,6 +317,53 @@ def get_sandbox_dataset_id(self):
         """
         return self._sandbox_dataset_id
 
+    def get_table_counts(self, dataset, tables, client=None):
+        """
+        Method to get the row counts of the list of tables
+
+        :param dataset: dataset identifier
+        :param client: big query client that has been instantiated
+        :param tables: list of tables
+        :return: returns a dictionary with table name as key and row count as value
+                counts_dict -> {'measurement' : 100000000, 'observation': 2000000000000}
+        """
+        counts_dict = dict()
+        for table in tables:
+            query = self.TABLE_COUNT_QUERY.format(dataset=dataset, table=table)
+            count = client.query(query).to_dataframe()
+            counts_dict[table] = count['row_count'][0]
+        return counts_dict
+
+    def validate_delete_rule(self, dataset, sandbox_dataset, sandbox_tables,
+                             tables_affected, initial_counts, client):
```

Comment:
I would really like to keep the client object out of the class cleaning rules.  That may / may not be possible.  I'm just trying to think through it  right now.

Thoughts?

Reply:
I thought of creating a client inside the method before adding it. but then I figured we shouldn't create too many clients in the workflow which will increase the running time. so I thought There will be a client instantiated in a cleaning rule so we can just use that.

I am open to using any way that we all can agree on.

Reply:
This is much better than creating the client in each class.  With this setup, the engine can pass the client it is already using.  

I'm just trying to think of a way to make these functions act like the cleaning rule themselves.  Can they just produce the sql needed and let that be executed by the clean engine?  I'm not sure if it can or can't.  If passing the sql to the engine can't work, then passing the client object from the engine to the validation methods is highly preferable to creating a new client in the validation methods.

Reply:
I tried to get it the way you were thinking, but it wouldn't be possible for this scenario because for a cleaning rule we are trying to collect all the queries and run all of them sequentially. the use case here is we are querying for some kind of a result either row_count or state of the data which needs to be saved to make the validation possible. as long as we have something to query and save before running the cleaning rule I couldn't think of a different approach.

Reply:
Ok.  So these methods are validation helper methods.  They are anticipated to be used by the ` setup_validation(self, *args, **keyword_args)` and ` validate_rule(self, *args, **keyword_args)` methods.  This means we expect to pass the `client` object from the engine into these methods (same for the `setup_rule(self, *args, **keyword_args)` method).  I don't expect what is passed from the engine into the class object to change.  What are your thoughts on updating these signatures to specifically include the `client` object, `(client, *args, **keyword_args)`?

Reply:
I think it is a good idea.